async_producer.go

```diff
@@ -218,7 +218,16 @@ func (p *asyncProducer) topicDispatcher() {
 
 		if msg.flags&shutdown != 0 {
```

Comment:
> There will only be one message with the shutdown message set, and it doesn't need to propagate further anymore, right?

I guess in theory there could be several if the user calls `AsyncClose` several times, but I don't feel like guarding against that case.

> Instead of sending a message with a flag, could we simply use a closing chan none for this now? Or am I missing something? I'd rather minimize the number of "special" messages that go through the pipeline

It would mean this goroutine would need a `select` instead of a nice simple `range`... not sure if that really ends up being any cleaner, I'll mock it quickly to see.